src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -4022,17 +4022,23 @@ public override IOperation VisitCaughtException(ICaughtExceptionOperation operat
 
         public override IOperation VisitInvocation(IInvocationOperation operation, int? captureIdForResult)
         {
-            if (operation.Instance != null)
+            IOperation instance = operation.TargetMethod.IsStatic ? null : operation.Instance;
+            (IOperation visitedInstance, ImmutableArray<IArgumentOperation> visitedArguments) = VisitInstanceWithArguments(instance, operation.Arguments);
+            return new InvocationExpression(operation.TargetMethod, visitedInstance, operation.IsVirtual, visitedArguments, semanticModel: null, operation.Syntax,
```

Comment:
>semanticModel [](start = 124, length = 13)

Why do we not add a semantic model here?

Reply:
> Why do we not add a semantic model here? 

We intentionally remove SemanticModel from all nodes. Right now it is used only to lazily calculate Parent for a node based on original IOperation tree and syntax, this won't work with rewritten tree and shouldn't even be attempted. We are using SemanticModel in this rewriter, but it won't be possible to feed any of the rewritten nodes back into this rewriter.  

---
In reply to: [188160319](https://github.com/dotnet/roslyn/pull/26797#discussion_r188160319) [](ancestors = 188160319)